src/BugsnagUnity/Native/Fallback/CacheManager.cs

```diff
@@ -166,32 +166,32 @@ private void RemoveOldestFiles(string[] filePaths, int numToRemove)
 
         public void RemoveCachedEvent(string id)
         {
-            try
+            RemovePayloadWithID(GetCachedEventFiles(), id);
+        }
+
+        public void RemoveCachedSession(string id)
+        {
+            RemovePayloadWithID(GetCachedSessionFiles(), id);
+        }
+
+        private void RemovePayloadWithID(string[] files, string id)
+        {
+            foreach (var path in files)
             {
-                foreach (var cachedEventPath in GetCachedEventFiles())
+                if (path.Contains(id))
                 {
-                    if (cachedEventPath.Contains(id))
-                    {
-                        File.Delete(cachedEventPath);
-                    }
+                    DeleteFile(path);
+                    return;
                 }
             }
-            catch { }
         }
 
-        public void RemoveCachedSession(string id)
+        private void DeleteFile(string path)
```

Comment:
I see the idea behind this method, but it's only called from one place currently.  Did you mean to move the other instances of `File.Delete` across to use it (and remove their try/catch blocks)?

Reply:
done and added a similar WriteFile method to reduce try catch blocks 👍